road/stats.py

```diff
@@ -0,0 +1,306 @@
+import os
+import pandas as pd
+import numpy as np
+import json
+import yaml
+
+
+def prepare_time_period_dfs(est_df, obs_df, times, combined_df_cols):
+    error = {col: est_df[col] - obs_df[col] for col in times}
+    error_df = pd.DataFrame(error)
+    error_squared_df = error_df.pow(2)
+    error_percent_df = error_df.divide(obs_df[times]).fillna(0)
+    error_percent_df = error_percent_df.apply(lambda x: x * 100, axis=1)
+
+    combined_df = pd.concat(
+        [est_df, obs_df[times], error_df, error_squared_df, error_percent_df], axis=1)
+    combined_df = combined_df.drop_duplicates(subset=['A', 'B'], keep='first')
+
+    calculation_cols = [
+        'Estimated Volume',
+        'Observed Volume',
+        'Errors',
+        'Squared Errors',
+        'Percent Errors']
+
+    time_period_dfs = {}
+
+    for period in times:
+        matching_column_indices = [
+            i for i, col_name in enumerate(
+                combined_df.columns) if col_name == period]
+        select_time_period_df = combined_df.iloc[:, matching_column_indices]

```

Comment:
Can't we select the columns directly here, and no need to rely on the column number (i)?
something like (please verify): `combined_df[period]`?
(I'm not quite following the logic in the creation of matching_column_indices: you're selecting only the column names where col_name == period. So that means you just want the indices (shouldn't there only be one?) of the column with the name period. So we can just select the "period" column directly, right?)